### PR TITLE
krew 0.3.2 (new formula)

### PR DESCRIPTION
**Pull request by glensc**
For now, install pre-compiled binary distribution.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by dawidd6** [comment]
You can try this:
```
go build -o bin/"krew" ./cmd/krew/...
```

**Comment by SMillerDev** [comment]
Binary distributions are not allowed. Formula have to build from source as per https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae

**Comment by glensc** [comment]
thanks, both, I first wanted to test does packaging even work. but I can't mark PR as WIP after it's been created in github.

for me somehow the `kubectl krew` command does not work.

**Comment by glensc** [comment]
tried source builds, different variations based on other formula builds, I keep ending up complaints of  `go.mod`: 5a2475ab70

```
==> gox build -tags netgo -mod readonly -output /usr/local/Cellar/krew/0.3.2/bin/krew
Last 15 lines from /Users/glen/Library/Logs/Homebrew/krew/01.gox:
2019-11-25 15:21:35 +0200

gox
build
-tags
netgo
-mod
readonly
-output
/usr/local/Cellar/krew/0.3.2/bin/krew

```

based on https://github.com/kubernetes-sigs/krew/blob/b303e3ae0179431750deab95be05e7b163286c74/hack/make-binaries.sh

**Comment by dawidd6** [comment]
Don't use `gox`. It's just an utility that uses `go` under the hood with env variables and options set to build the project for multiple targets at once. We only need one target.

Try this:
```ruby
  def install
    ENV["CGO_ENABLED"] = "0"
    system "go", "build", "-o", bin/"krew", "-tags", "netgo", "./cmd/krew/..."
  end
```

**Comment by glensc** [comment]
@dawidd6 I tried pure `go` first, end result they failed the same way. will test your code suggestion.

**Comment by glensc** [comment]
your given command produced this error:

```
==> go build -o /usr/local/Cellar/krew/0.3.2/bin/krew -tags netgo ./cmd/krew/...
Last 15 lines from /Users/glen/Library/Logs/Homebrew/krew/01.go:
go: finding github.com/sahilm/fuzzy v0.0.5
go: finding github.com/spf13/cobra v0.0.3
go: finding github.com/spf13/pflag v1.0.3
go: finding github.com/mattn/go-colorable v0.1.2
go: finding k8s.io/client-go v7.0.0+incompatible
go: finding k8s.io/apimachinery v0.0.0-20190717022731-0bb8574e0887
go: finding sigs.k8s.io/yaml v1.1.0
go: finding gopkg.in/yaml.v2 v2.2.2
go: finding github.com/gogo/protobuf v1.2.1
go: finding github.com/google/gofuzz v1.0.0
go: finding k8s.io/klog v0.3.3
go: finding gopkg.in/inf.v0 v0.9.1
go: finding golang.org/x/net v0.0.0-20190628185345-da137c7871d7
go: finding golang.org/x/text v0.3.2
go build: cannot write multiple packages to non-directory /usr/local/Cellar/krew/0.3.2/bin/krew
```

even it created single binary `krew`:

```ruby
    bin.mkpath
    system "go", "build", "-tags", "netgo", "./cmd/krew/..."
```

so, installing to build dir and then install.

**Comment by glensc** [comment]
works now locally:

```
[/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core (add-krew)] ➔ kubectl krew update
Updated the local copy of plugin index.

[/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core (add-krew)] ➔ kubectl krew search krew
NAME  DESCRIPTION                           INSTALLED
krew  Package manager for kubectl plugins.  no
[/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core (add-krew)] ➔
```

File: Formula/krew.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  depends_on "kubernetes-cli"
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    # build in local dir to avoid this error:
+    # go build: cannot write multiple packages to non-directory /usr/local/Cellar/krew/0.3.2/bin/krew
+    mkdir "build"
+    system "go", "build", "-o", "build", "-tags", "netgo", "./cmd/krew/..."
+    # install as kubectl-krew for kubectl to find as plugin
+    bin.install "build/krew" => "kubectl-krew"
```
Comment (chenrui333) [inline Formula/krew.rb]: Just curious, why do you prefer `kubectl-krew` vs `krew`?

Comment (glensc) [reply Formula/krew.rb]: I don't prefer it, it's how kubectl finds its plugins, and the comment above also indicates that.

Comment (chenrui333) [reply Formula/krew.rb]: I see, thanks for the context!

**Review by chenrui333 — APPROVED** [review]
lgtm

**Comment by chenrui333** [comment]
@glensc Thanks a lot for your contribution!!

**Pull request closed without merging**